pkg/adminconfig/evaluator_interfaces.md

```diff
@@ -0,0 +1,259 @@
+
+The mechanism of tracking the changes in infrastructure and policies is TBD. In the initial approach, infrastructure and modules are loaded once.
+
+After the query is evaluated, it is parsed into a list of decisions per capability. Then, all decisions on the same capability are merged into one. For example, a decision to deploy read at the workload scope and a decision to deploy read in the workload cluster will result in a single decision to deploy read at the workload scope in the workload cluster. A decision to deploy a capability in clusterA, merged with a decision to deploy this capability in 
+any available cluster, will result in a decision to deploy in clusterA.
+
+If the merge process does not succeed to provide a consistent solution, `Valid` will be set to false in `EvaluatorOutput`, and plotter generation will fail.
+
+### Policies
+
+#### Syntax
+
+Policies are written in rego files. Each file declares a package `adminconfig.<name>`, e.g. `adminconfig.default_policies`
+
+Rules are written in the following syntax: `config[{capability: decision}]` where
+
+`capability` represents a required module capability, such as "read", "write", "transform" and "copy".
+
+`decision` is a JSON structure that matches `Decision` defined above. 
+
+```
+{ 
+	"policy": {"ID": <id>, "description": <description>}, 
+	"deploy": <true, false>,
+	"restrictions": {
+		"modules": <map {key, value}>,
+		"clusters": <list of cluster names>,
+	},
+}
+```
+
+#### Mandatory policies
+
+Mandatory policies come with the fybrik deployment. They define the deployment of basic capabilities, such as read and write. 
+Mandatory policies reside in /tmp/adminconfig/ directory of the manager pod. 
+Any other policy package will be loaded in addition to these policies.
+
+The package below defines the mandatory policies.
+
+```
+package adminconfig.default_policies
+
+config[{"read": decision}] {
+    read_request := input.request.usage.read
+    policy := {"ID": "read-default", "description":"Read capability is requested for read workloads"}
+    decision := {"policy": policy, "deploy": read_request}
+}
+
+config[{"write": decision}] {
+    write_request := input.request.usage.write 
+    policy := {"ID": "write-default", "description":"Write capability is requested for workloads that write data"}
+    decision := {"policy": policy, "deploy": write_request}
+}
+```
+
+#### Extended policies
+
+The extended policies define advanced deployment requirements, such as where read or transform modules should run, what should be the scope of module deployments, and more. 
+
+In the initial approach, the extended policies will be loaded during the control plane deployment, and will reside in /tmp/adminconfig/ together with the mandatory policies.
+
+In the future, the extended policies will be loaded dynamically via a configmap.
```

Comment:
The IT admin will indicate which policies should be loaded via the configmap?  That the intention?

Reply:
IT admin will be able to load policies after Fybrik is up and running. This will be done using a configmap, as with governance policies.